b2sdk/scan/folder.py

```diff
@@ -215,26 +213,19 @@ def _walk_relative_paths(
         # This is because in Unicode '.' comes before '/', which comes before '0'.
         names = []  # list of (name, local_path, relative_file_path)
 
-        #FIXME should I construct starting_dir dynamically based on relative_dir_path and local_dir?
-        if starting_dir is None:
-            starting_dir = local_dir
-
         if visited_symlinks is None:
             visited_symlinks = set()
 
         if local_dir.is_symlink():
             real_path = local_dir.resolve()
-
-            if real_path in starting_dir.parents:
-                # Skip symlink if it points to an ancestor directory of the initial scan directory
-                #FIXME I think I should raise an error here instead of skipping the symlink
-                return
-
-            if real_path in visited_symlinks:
-                # Infinite symlink loop detected, skip symlink
+            inode_number = real_path.stat().st_ino
```

Comment:
@ppolewicz As per our discussion, I changed the approach to reduce memory usage. Assuming:

- 64-bit system, with 28 bytes per integer
- 1 billion directories in the scanned directory tree
-  5% of these are symlinks (50m)

The pessimistic size of the visited_symlinks would be around 1.4GB. I would argue this is acceptable, but please share your thoughts.

we are on 64-bit system and an integer takes 28 bytes, we can calculate have the following: 1 billions folders in the traversed tree, with